internal/sudo/contract_manager.go

```diff
@@ -75,37 +74,72 @@ type OpenAckDetails struct {
 	CounterpartyVersion   string `json:"counterparty_version"`
 }
 
+type ContractMethods interface {
+	HasContractInfo(sdk.Context, sdk.AccAddress) bool
+	Sudo(sdk.Context, sdk.AccAddress, []byte) ([]byte, error)
+}
+
+var _ ContractMethods = (*ContractManager)(nil)
+
+type ContractManager struct {
+	wasmKeeper ContractMethods
+}
+
 type Handler struct {
```

Comment:
wdyt about splitting the file for a better readability into three files:
1. all models e.g. results, responses etc.;
2. ContractManager + ContractMethods;
3. Handler.

or at least move the Handler definition and its methods after the ContractManager's ones?

Reply:
do we even need both Handler and ContractManager?
1. they are very much coupled to each other (ContractManager provides a constructor for the Handler, and the constructor requires a ContractManager);
2. ContractManager is very thin and it mostly proxies requests to a wasm.Keeper.

what you think about merging them into a single struct that will:
1. have access to chain contracts through a wasm.Keeper;
2. call sudo to these contracts.

that could simplify structure a lot imo

Reply:
> wdyt about splitting the file for a better readability into three files:
> ...

Good point.

Reply:
> 1. they are very much coupled to each other (ContractManager provides a constructor for the Handler, and the constructor requires a ContractManager);

Yes, I will decrease coupling by adding `NewSudoHandler` to the `ContractMethods` interface.

> 2. ContractManager is very thin and it mostly proxies requests to a wasm.Keeper.
That's was the idea of Contract Manager to reduce dependency from wasm.Keeper. It inverts control from keeper to interface and interface realization.